nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -89,9 +134,51 @@
         .defaultValue(StorageClass.Standard.name())
         .build();
 
+    public static final PropertyDescriptor MULTIPART_THRESHOLD = new PropertyDescriptor.Builder()
+            .name("Multipart Threshold")
+            .description("Specifies the file size threshold for switch from the PutS3Object API to the " +
+                    "PutS3MultipartUpload API.  Flow files bigger than this limit will be sent using the stateful " +
+                    "multipart process.\n" +
+                    "The valid range is 50MB to 5GB.")
+            .required(true)
+            .defaultValue("5 GB")
+            .addValidator(StandardValidators.createDataSizeBoundsValidator(MIN_S3_PART_SIZE, MAX_S3_PUTOBJECT_SIZE))
+            .build();
+
+    public static final PropertyDescriptor MULTIPART_PART_SIZE = new PropertyDescriptor.Builder()
+            .name("Multipart Part Size")
+            .description("Specifies the part size for use when the PutS3Multipart Upload API is used.\n" +
+                    "Flow files will be broken into chunks of this size for the upload process, but the last part " +
+                    "sent can be smaller since it is not padded.\n" +
+                    "The valid range is 50MB to 5GB.")
+            .required(true)
+            .defaultValue("5 GB")
+            .addValidator(StandardValidators.createDataSizeBoundsValidator(MIN_S3_PART_SIZE, MAX_S3_PUTOBJECT_SIZE))
+            .build();
+
+    public static final PropertyDescriptor MULTIPART_S3_AGEOFF_INTERVAL = new PropertyDescriptor.Builder()
+            .name("Multipart Upload AgeOff Interval")
+            .description("Specifies the interval at which existing multipart uploads in AWS S3 will be evaluated " +
+                    "for ageoff.  Calls to onTrigger() will initiate the ageoff evaluation if this interval has been " +
```

Comment:
I think that that this should be "When processor is triggered" rather than "calls to onTrigger()" to prevent too much java'ism leaking out.